doc/hardware/porting/board_porting.rst

```diff
@@ -34,39 +88,23 @@ You can visualize the hierarchy like this:
 Here are some examples. Notice how the SoC series and family levels are
 not always used.
 
-.. list-table::
-   :header-rows: 1
-
-   * - Board
-     - SoC
-     - SoC series
-     - SoC family
-     - CPU core
-     - Architecture
-   * - :ref:`nrf52dk_nrf52832 <nrf52dk_nrf52832>`
-     - nRF52832
-     - nRF52
-     - Nordic nRF5
-     - Arm Cortex-M4
-     - Arm
-   * - :ref:`frdm_k64f <frdm_k64f>`
-     - MK64F12
-     - Kinetis K6x
-     - NXP Kinetis
-     - Arm Cortex-M4
-     - Arm
-   * - :ref:`stm32h747i_disco <stm32h747i_disco_board>`
-     - STM32H747XI
-     - STM32H7
-     - STMicro STM32
-     - Arm Cortex-M7
-     - Arm
-   * - :ref:`rv32m1_vega_ri5cy <rv32m1_vega>`
-     - RV32M1
-     - (Not used)
-     - (Not used)
-     - RI5CY
-     - RISC-V
+.. table::
+
+   +--------------------------------------------------+-------------+-------------+---------------+---------------+----------------+--------------+
+   | Board                                            | Identifier  | SoC         | SoC Series    | SoC family    | CPU core       | Architecture |
```

Comment:
> It would be good to have an example with more than just a soc or CPU selection (meaning an example with a SOC + CPU + variant selection)

I think the variant selection part goes a bit beyond the purpose of the original table, which is seen here: https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#boards-socs-etc

where I believe the purpose is to show the different layers.
Those layers don't change when building for a variant.

My main purpose with updating the table is to show how the identifier fits into this table, and the fact that a board now supports multiple SoCs or cpu-clusters (often a cpu-cluster in identical to a cpu core)

And as part of this, show that two cores may be of different architectures.

> The "CPU core" column seems a bit ambiguous to me. Looking at the current description I do not if you mean the "CPU model/type" or you intend to refer to a particular CPU instance. 

I intended to extend the table to show the cpu-cluster as we describe it in the soc.yml, as that is what users will be building for.
So in the nrf5340 case, this would be `cpuapp` and `cpunet`, both of type cortex-m33.

> An example with a AMP SOC with the same type of core would be nice

Guess the nrf5340 would be a good candidate.
Probably a dual SoC board should also be added to this table.

Reply:
Added nRF5340 to the table.
Atm it's the only amp soc on the collab-hwm branch which has identical architecture on both cores.